src/shared/common/vector_functions.cpp

```diff
@@ -3,6 +3,21 @@
 namespace SPH
 {
 	//=================================================================================================//
+     Vec3d OcctToEigen(const gp_Pnt &occt_vector)
+   {
+    return Vec3d(occt_vector.X(), occt_vector.Y(), occt_vector.Z());
+   }
+   //=================================================================================================//
+     gp_Pnt EigenToOcct(const Vec3d &eigen_vector)
+    {
+    return gp_Pnt(eigen_vector[0], eigen_vector[1], eigen_vector[2]);
+    }
+   //=================================================================================================//
+   Vec3d OcctVecToEigen(const gp_Vec &occt_vector)
+   {
+    return Vec3d(occt_vector.X(), occt_vector.Y(), occt_vector.Z());
+   }
+	//=================================================================================================//
```

Comment:
Please move things out of `src/shared` to keep occt bounded out of the core lib at the moment

Reply:
> Please move things out of `src/shared` to keep occt bounded out of the core lib at the moment

Yes i have removed things to the extra source part